src/primitives/transaction.h

```diff
@@ -127,13 +127,72 @@ class CTxIn
     std::string ToString() const;
 };
 
+
+/**
+ * CAmount is an int64_t, which requires alignment of 8. This wrapper stores the data
+ * in a memory array and packs/unpacks it whenever needed, regarldess of alignment.
+ *
+ * Since CAmount's size is know, the compiler should be able to optimize the std::memcpy away
+ * in most cases.
+ */
+class PackableCAmount
+{
+    uint8_t m_data[sizeof(CAmount)];
+
+public:
+    PackableCAmount(CAmount val) noexcept
```

Comment:
Should be `explicit`? :)

Reply:
It should, I can add that easily by adding the assignment operator for CAmount. 

Also theoretically the `operator CAmount()` should be explicit too, but changing that would mean that I'd have to change practically all usages of this class...